src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -331,7 +331,6 @@ public override SecurityKey SigningKey
         /// <param name="rawData">the original token.</param>
         private void Decode(string[] tokenParts, string rawData)
         {
-            LogHelper.LogInformation(LogMessages.IDX14106, rawData);
```

Comment:
We log exceptions with rawData [here](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/9170aa6db120342ab65c9ea7e045b291e8fef908/src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs#L340:~:text=throw%20LogHelper.LogExceptionMessage(new%20ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX14102%2C%20tokenParts%5B0%5D%2C%20rawData)%2C%20ex))%3B) and [here](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/9170aa6db120342ab65c9ea7e045b291e8fef908/src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs#L340:~:text=throw%20LogHelper.LogExceptionMessage(new%20ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX14101%2C%20tokenParts%5B1%5D%2C%20EncodedToken)%2C%20ex))%3B) if we're unable to decode/deserialize the header or payload.

Reply:
Is there value in logging the encoded header and payload in [DecodeJws(string[] tokenParts)](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/9170aa6db120342ab65c9ea7e045b291e8fef908/src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs#:~:text=private%20void%20DecodeJws(string%5B%5D%20tokenParts))?

Reply:
According to the JWE documentation, the token is encrypted and so it doesn't make sense to log it. Is my understanding correct?

Reply:
This seems largely right to me. On the other hand logging the encrypted token seems less risky since presumably someone would have to have to key to access the PII we're attempting to protect. Seems plausible there might be some cases where the full token doesn't decrypt but the header (which I understand has to be unencrypted) would still provide diagnostically useful info.